Added FreeBSD aarch64 (ARM64) support

siepkes opened the pull request:
Tested on a Raspberry Pi 4.

MrDOS commented on src/main/java/gnu/io/NativeResource.java:
```diff
@@ -186,7 +186,11 @@ private InputStream locateResource(String name) {
 			}
 		}else if(OSUtil.isFreeBSD()) {
-			if(OSUtil.is64Bit()) {
+			if(OSUtil.isARM()) {
+				if(OSUtil.is64Bit()) {
```
Could just be

```suggestion
			if(OSUtil.isARM() && OSUtil.is64Bit()) {
```

MrDOS commented on src/main/c/Makefile:
```diff
@@ -239,6 +239,13 @@ freebsd64: export platform := freebsd/x86_64
 	$(MAKE) -f natives.mk
 
+# Requires a FreeBSD host running on aarch64 (arm64) architecture.
+freebsdarm64v8: export CFLAGS += -I$(JAVA_HOME)/include/freebsd -m64
+freebsdarm64v8: export LDFLAGS += -m64
```
I think [the `-m64` flag](https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html#index-m64-5) is AMD64-specific, and I suspect it's ignored outright when building for AArch64. Have you tried without the flag to confirm it's necessary, or if not, can you please do so?

siepkes replied:
Your right! I missed that one. I have removed the flag and verified the resulting binary still works.

MrDOS reviewed (changes requested):
Thank you for your contribution! This looks good. As noted below, can you confirm that the `-m64` flag is strictly necessary? Then I'm happy to merge.

My biggest reservation is that native libraries for all of the other supported platforms can be cross-compiled, and there's no off-the-shelf cross-compiler to target FreeBSD arm64 from Linux amd64. However, the actual build process integrates cleanly with the others, and it looks possible to extend [the current FreeBSD cross-compilation environment](https://github.com/MrDOS/freebsd-cross-build) to support arm64, so I'm happy to accept this as un-CI'd for now with the expectation that I'll be able to add it to CI later.

siepkes commented:
Thanks for the prompt review! I've addressed your feedback. On closer inspection I also noticed I needed to add the variant in the makefile so the resulting binary has the correct name.

You might already know about it but there is also the [Maven NAR plugin](https://github.com/maven-nar/nar-maven-plugin) to aid in dealing with native code.

MrDOS commented on src/main/c/Makefile:
```diff
@@ -239,6 +239,13 @@ freebsd64: export platform := freebsd/x86_64
 	$(MAKE) -f natives.mk
 
+# Requires a FreeBSD host running on aarch64 (arm64) architecture.
+freebsdarm64v8: export CFLAGS += -I$(JAVA_HOME)/include/freebsd -m64
```
You've removed the `-m64` linker flag; did you try without the compiler one, too?

siepkes replied:
Sorry, I totally missed the compiler flag. I either need more caffeine or more vacation it seems ;-)

MrDOS commented on src/main/c/Makefile:
```diff
@@ -239,6 +239,13 @@ freebsd64: export platform := freebsd/x86_64
+# Requires a FreeBSD host running on aarch64 (arm64) architecture.
+freebsdarm64v8: export CFLAGS += -I$(JAVA_HOME)/include/freebsd -m64
+freebsdarm64v8: export platform := freebsd/ARM_64
+freebsdarm64v8: export variant := v8
```
Ah, right, because `gnu.io.NativeResource.loadLib()` will always try to load a library called `libNRJavaSerialv8` for 64-bit ARM. Good catch.

MrDOS commented on src/main/java/gnu/io/NativeResource.java:
```diff
@@ -186,7 +186,9 @@ private InputStream locateResource(String name) {
 			}
 		}else if(OSUtil.isFreeBSD()) {
-			if(OSUtil.is64Bit()) {
+			if(OSUtil.isARM() && OSUtil.is64Bit()) {
```
Thank you for refactoring this condition. I know it's stylistically at odds with the other platform conditions, but IMO it reads better for a platform with only three options. I'd like to replace these condition trees with a table-based lookup anyway.

siepkes commented:
Alright, took some tries but I think we've got there! Thanks for your patience!

Building with the `-m64` compiler flag removed yields exactly the same binary (I compared the hashes).

MrDOS reviewed (approved):
Thank you for being willing to iterate on this. (Thank you also for keeping a neat and tidy squashed history.) I'm quite happy with this now, and glad to merge it.
